conf/machine/baremetal-riscv32.conf

```diff
@@ -23,7 +23,7 @@ USER_CLASSES_remove = "image-prelink"
 #TARGET_LINK_HASH_STYLE = ""
 LINKER_HASH_STYLE = ""
 
-TCLIBC = "baremetal"
+TCLIBC ?= "baremetal"
```

Comment:
we may want to define another machine config, baremetal is for generating pure baremetal toolchain

Reply:
Same point as above, it's confusing when the override doesn't work

Reply:
baremetal is baremetal. I dont think musl/glibc argument applies here.  Probably its better to post a new .conf file which takes care of other TCLIBC options. e.g. newlib

Reply:
Ok, I'll leave this as is for now.